src/communication/request_sync_client_server.hpp

```diff
@@ -0,0 +1,111 @@
+#ifndef REQUEST_SYNC_CLIENT_SERVER_HPP
+#define REQUEST_SYNC_CLIENT_SERVER_HPP
+
+#include "request_client_server.hpp"
+#include "request_sync_send_queue.hpp"
+
+//
+// declarations
+//
+namespace cubcomm
+{
+  /* wrapper/helper encapsulating instances of request_client_server, request_sync_send_queue
+   * and request_queue_autosend working together
+   */
+  template <typename T_OUTGOING_MSG_ID, typename T_INCOMING_MSG_ID, typename T_PAYLOAD>
+  class request_sync_client_server
+  {
+    public:
+      using request_client_server_t = cubcomm::request_client_server<T_OUTGOING_MSG_ID, T_INCOMING_MSG_ID>;
+      using incoming_request_handler_t = typename request_client_server_t::server_request_handler;
+
+    public:
+      request_sync_client_server (cubcomm::channel &&a_channel,
+				  std::map<T_INCOMING_MSG_ID, incoming_request_handler_t> &&a_incoming_request_handlers);
+      ~request_sync_client_server ();
+      request_sync_client_server (const request_sync_client_server &) = delete;
+      request_sync_client_server (request_sync_client_server &&) = delete;
+
+      request_sync_client_server &operator = (const request_sync_client_server &) = delete;
+      request_sync_client_server &operator = (request_sync_client_server &&) = delete;
+
+    public:
+      bool is_connected () const;
+
+      /* only used by unit tests
+       */
+      std::string get_underlying_channel_id () const;
+
+      void push (T_OUTGOING_MSG_ID a_outgoing_message_id,
+		 T_PAYLOAD &&a_payload);
+
+    private:
+      using request_sync_send_queue_t = cubcomm::request_sync_send_queue<request_client_server_t, T_PAYLOAD>;
+      using request_queue_autosend_t = cubcomm::request_queue_autosend<request_sync_send_queue_t>;
+
+    private:
+      std::unique_ptr<request_client_server_t> m_conn;
+      std::unique_ptr<request_sync_send_queue_t> m_queue;
+      std::unique_ptr<request_queue_autosend_t> m_queue_autosend;
+  };
+}
+
+//
+// implementations
+//
+namespace cubcomm
+{
+  template <typename T_OUTGOING_MSG_ID, typename T_INCOMING_MSG_ID, typename T_PAYLOAD>
+  request_sync_client_server<T_OUTGOING_MSG_ID, T_INCOMING_MSG_ID, T_PAYLOAD>::request_sync_client_server (
+	  cubcomm::channel &&a_channel,
+	  std::map<T_INCOMING_MSG_ID, incoming_request_handler_t> &&a_incoming_request_handlers)
+    : m_conn { new request_client_server_t (std::move (a_channel)) }
+  , m_queue { new request_sync_send_queue_t (*m_conn) }
+  , m_queue_autosend { new request_queue_autosend_t (*m_queue) }
```

Comment:
Intentation looks off
```suggestion
    , m_queue { new request_sync_send_queue_t (*m_conn) }
    , m_queue_autosend { new request_queue_autosend_t (*m_queue) }
```

Reply:
AStyle is fooled by the curly brace initialization.

Reply:
Damn the fool :)).

Reply:
Changing back to brackets works. I suppose is just not implemented.